retrofit/src/main/java/retrofit/Endpoint.java

```diff
@@ -2,29 +2,13 @@
 
 import com.squareup.okhttp.HttpUrl;
 
-import static retrofit.Utils.checkNotNull;
-
 /** An API endpoint. */
-public abstract class Endpoint {
-  /** Create an endpoint with the provided {@code url}. */
-  public static Endpoint createFixed(String url) {
-    checkNotNull(url, "url == null");
-    final HttpUrl httpUrl = HttpUrl.parse(url);
-    if (httpUrl == null) {
-      throw new IllegalArgumentException("Invalid URL: " + url);
-    }
-    return new Endpoint() {
-      @Override public HttpUrl url() {
-        return httpUrl;
-      }
-    };
-  }
-
+public interface Endpoint {
```

Comment:
Yeah it needs to be able to change over time. We use it internally and there's a few external consumers of the behavior.